code/client/munkilib/updatecheck.py

```diff
@@ -1030,6 +1030,25 @@ def compare_version_keys(a, b):
                         rejected_items.append(reason)
                         continue
 
+                if item.get('deploy_percent', ''):
+                    deploy_percent = item['deploy_percent']
+                    munkicommon.display_debug1(
+                        'Considering item %s, version %s '
+                        'with %s deploy percent required',
+                        item['name'], item['version'], deploy_percent)
+                    munkicommon.display_debug1(
+                        'Machine deploy percent is %s', MACHINE['shard'])
+                    if (MACHINE['shard'] > deploy_percent):
+                        # skip this one, go to the next
+                        munkicommon.display_debug1(
+                            'Item %s, version %s is set to deploy to machines '
+                            'with a shard value of %s or lower. '
+                            'Our shard value is %s.'
+                            % (item['name'], item['version'],
```

Comment:
Great. Now maybe a consistent terminology in lines 1036-1040. I think shard vs deploy_percent will be confusing.

Maybe even change the name of the "deploy_percent" key to "minimum_shard_value" or something similar.